Assets/Scripts/ViconNexusUnityStream/DataStreamer.cs

```diff
@@ -60,14 +82,24 @@ private void StreamData(byte[] receivedData)
             rawData[subject] = JsonConvert.SerializeObject(data);
         }
     }
-    
-    private void FixedUpdate()
+
+    /// <summary>
+    /// Regsiter a subject to recieve subject data.
+    /// </summary>
+    public void RegisterSubject(string subjectName)
     {
-        webSocket.DispatchLatestMessage();
+        subjectList.Add(subjectName);
+        if (webSocket == null || webSocket.State == WebSocketState.Closed || webSocket.State == WebSocketState.Closing)
+        {
+            SetupConnection();
```

Comment:
(logically it makes sense but i have a bad feeling about calling SetupConnection() here but its aight)

Reply:
why?

Reply:
tbh its an irrational ick

but i feel like RegisterSubject() should not automatically call SetupConnection() and that it should be handled in awake or start